### DISCUSSION TITLE
Detailed procedure/video on how to build cura from source in windows system,&ustom exe file of our own icon and naming not untitled - Ultimaker cura)on the top left of the window with log

### DISCUSSION THREAD

**Discussion by CHSIVASANTHOSH** [smartavionics/Cura#208] [Q&A]
Dear @smartavionics greetings of the day , i would be glad if you can make detailed procedure/video on how to build cura from source in windows system and how do we get our custom exe file of our own icon and naming so that on main window splash there won't be any ultimaker on it (untitled - Ultimaker cura)on the top left of the window with logo<--like this

**Comment by smartavionics** [comment]
I can't make a video as I don't use Windows for anything but building Cura releases and know nothing about it.

To build Cura from my repos you need to:

1 - install git

2 - git clone my cura-build-environment repo and check out the mb-master branch

3 - Look at the README.md file and install all the required stuff (cmake, microsoft visual studio, mingwin-w64).

4 - make a build sub-directory and open a command window, do "cd build", "..\run_cmake64.bat", "nmake". This will take many hours to build.

5 - git clone my cura-build repo and check out the 4.20 branch. Again make a build sub-directory and in there do "..\run_cmake64.bat", "nmake package". This will create the Cura installer.

Good luck!

**Reply by CHSIVASANTHOSH** [reply]
Thank you @smartavionics for your kind gesture towards my problem I will try your way but present I need to build 5.2.2 version may I know your suugested way can be suitable for it ,as its a high priority for me for now

**Comment by smartavionics** [comment]
Ah, in that case, I can't help you as the instructions above are for a Cura 4 based build. I know nothing about building Cura 5. Sorry.

**Reply by CHSIVASANTHOSH** [reply]
I mark ,can you help me in the case how to change the name of the .exe file/build .exe file such that the name from ultimaker cura at the top beside icon needed to be changed & how to change the add  the .pyc files in library.zip files so that the curaversion.py, applicationmetadata.py , curaapplication.py these contains the details and name of the software ,so i want to change these,welcomepages.py

**Reply by CHSIVASANTHOSH** [reply]
@smartavionics can you suggest for this case

**Comment by meltiseugen** [comment]
I wasn't sure if I should start a separate discussion thread or use this one. 

I wanted to ask if the above-mentioned steps would work for building Cura (and CuraEngine) locally on my macOS?
Was thinking that instead of running a Linux build in a VM, to try and make a local build for macOS.

**Comment by smartavionics** [comment]
In theory, yes. In reality, I don't believe you can make a working macOS build from my sources. It used to be possible for old versions of macOS but I don't think that is the case now.
